client/src/pages/Calendar.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import Alert from "react-bootstrap/Alert";
+
+const Calendar = () => {
+	const [bookingsData, setBookingsData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		setIsLoading(true);
+		fetch("/api/bookings?filter=2weeks")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the calendar data, Please try again"
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setBookingsData(data);
+				setError(null);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	}, []);
+
+	if (isLoading) {
+		return (
+			<div className="d-flex justify-content-center">
+				<div className="spinner-grow">
+					<span className="visually-hidden">Loading...</span>
+				</div>
+			</div>
+		);
+	}
+	if (error != null) {
+		return <Alert variant="danger">{error}</Alert>;
+	}
+
+	const pickUpDates = [];
+
+	bookingsData.forEach((client) => {
+		let found = pickUpDates.find((el) => el.date == client.pick_up_date);
```

Comment:
```suggestion
		const existingPickUpDate = pickUpDates.find((el) => isSameDate(el.date, client.pick_up_date));
```
- In the codebase we're using const by default, so let's do that here
- We could try using a more descriptive name like `existingPickUpDate`
- Doing `el.date == client.pick_up_date` will check if the date **and** time are the same, but here we want to check if only the dates are the same. By checking only the dates, we are able to group the clients by date. We can do the comparison in a new function, where the comparison is `new Date(dateA).toDateString() === new Date(dateB).toDateString()`